drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/ScenarioSimulationServiceImpl.java

```diff
@@ -368,8 +370,17 @@ protected Package getOrCreateJunitActivatorPackage(KieModule kieModule) {
         return junitActivatorPackage;
     }
 
-    protected void removeOldActivatorIfExists(KieModule kieModule) {
+    /**
+     * This routine looks for existing activators to migrate
+     * @param activatorPath
+     * @param kieModule
+     */
+    protected void removeOldActivatorIfExists(org.uberfire.java.nio.file.Path activatorPath, KieModule kieModule) {
 
+        // migration step after Test Scenario runner modules split DROOLS-3389
+        ioService.deleteIfExists(activatorPath);
```

Comment:
@danielezonca 
This is related somehow to @jomarko doubts: expanding the code from _createActivatorIfNotExist_ and _removeOldActivatorIfExists_ we get the following (pseudo) code:
` if ( !ioService.exists(activatorPath)) ->   ioService.deleteIfExists(activatorPath);`

Maybe is needed/usefull because of other usage, but it actually looks _strange_

Reply:
I agree it could be a misleading but the logic is a bit different, the external condition is
`if (needMigrateActivatorIfExists || !ioService.exists(activatorPath))`
So the other method with the `deleteIfExists` can be invoked if the file doesn't exist (= useless call) but also if the variable `needMigrateActivatorIfExists` is true.
That variable is true if the project contains any "old dependencies".

The general problem is we should have a migration mechanism also for dependencies and activator but we have not. There are four possible statuses:
- empty project: nothing to migrate, just add dependencies and activator
- first version of activator: need to find where the activator is and remove it (now is in `testscenario` package)
- second version of activator: need to update dependencies and activator too
- current version: nothing to migrate

The general idea in the code is:
- first update/add/remove dependencies
- if I had to update the dependencies or the activator does not exist try to remove existing activator (it could be in different paths)
- create the new activator